### ISSUE TITLE
Get packaged into Conan

### ISSUE THREAD

**Issue by dankamongmen** [dankamongmen/notcurses#1098]
Conan is one of several emerging C/C++ "package managers" ala those used by node, go, and rust. We should get Notcurses in there early (NCURSES is already present):

https://conan.io/center/
https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md

**Comment by dankamongmen** [comment]
I've been added to the Conan EAP by @jgsogo. He's a good man, and thorough.

**Comment by dankamongmen** [comment]
https://github.com/dankamongmen/conan-center-index/tree/dankamongmen/notcurses has some progress.

**Comment by dankamongmen** [comment]
hey @grendello just wondering if you've ever wrapped something up for conan

**Comment by grendello** [comment]
Nope, I prefer to use direct dependencies in the form of distro provided packages. At least on Linux, which has a number of very decent packaging solutions. Conan (or others) would be useful if we want to target distros which have no decent package management system (Slack...?) or for Windows. For macOS (when it works) we'll probably want to write a Homebrew recipe (I did that for other software before) and be done with it.

**Comment by jgsogo** [comment]
Hi! 

Using dependencies installed in the system via any system package manager is suitable if there aren't conflicts or you don't want to test the library/application using different versions of those dependencies (assuming those dependencies can be installed system-wide).

With Conan (or _any_ other package manager) it is really easy to override dependencies in the graph and check if they work as expected... or to cross-compile for other architectures, or use different compiler versions. Basically it makes it possible to isolate the project from the running machine, dependencies of this project won't interfere with any other running application or any other project being developed in the same machine.

Of course, you get these advantages paying a bit of complexity... but IMHO, it is worth it.

**Comment by jgsogo** [comment]
That's really good feedback, @grendello. It is possible to _proxy_ a system library using a Conan recipe, but those are not the kind of Conan recipes that are gathering attention from our community right now. I totally agree Conan et al. are more useful for development than distribution right now.

**Comment by dankamongmen** [comment]
@jgsogo i'm definitely looking to get a conan recipe up and going. i spent two hours or so on it back about a month ago, and put it aside as it seemed more complex than any other packaging i'd been doing. i fully intend to get back on it, though i would absolutely love for someone else to do it, too =].

**Comment by jgsogo** [comment]
If you submit a PR with a first version of the recipe to https://github.com/conan-io/conan-center-index we can help you with the details, or we can merge it if it is good enough and then improve it in the future. We have a very active community of users 😺 .

**Comment by dankamongmen** [comment]
> If you submit a PR with a first version of the recipe to https://github.com/conan-io/conan-center-index we can help you with the details, or we can merge it if it is good enough and then improve it in the future. We have a very active community of users  .

i will take advantage of this friendly offer today!

**Comment by dankamongmen** [comment]
I now have a recipe which builds notcurses and a test program, and successfully launches the latter. We `require` "libunistring/0.9.10" and "openimageio/2.2.7.0" (i didn't see ffmpeg/libav). i'm gonna go ahead and submit this, and get some feedback.

**Comment by dankamongmen** [comment]
Just did a bunch of work on our conan recipe, and we're almost there. Just need to break libnotcurses and libnotcurses++ into different Components, and we ought be done.

**Comment by dankamongmen** [comment]
We might as well wait for 2.1.6 to drop, since we'll need a third Component for `notcurses-core` at that point. Once 2.1.6 is cut, though, let's add the two new Components and request a fresh review.

**Comment by dankamongmen** [comment]
ok @jgsogo , i've got two problems left:

* I need component `tinfo` from recipe `ncurses`, **not** full `ncurses`. linking with the latter will fail. it seems i ought be able to `require` `ncurses.tinfo` to do this, but it always fails. any suggestions?
* it doesn't want me using `import` in `test_package/conanfile.py`, but i need that for `os`, and all the other `conanfile.py`s through the tree seem to be doing this. what's up?

**Comment by dankamongmen** [comment]
went ahead and updated our outstanding patch to 2.3.1, and let's see what happens.
